src/OrderInventory.ts

```diff
@@ -0,0 +1,103 @@
+import { genericCleanup } from "./util";
+
+/**
+ * Sort Inventories by Code
+ * TODO: only grab the inventory where the button was pressed
+ */
+export class OrderInventory {
+  private sortByCode: boolean = false;
+  private tag = "sort-by-code";
+  private cleanups: Array<() => void> = [];
+
+  cleanup() {
+    this.cleanups.forEach((f, i) => {
+      f();
+      delete this.cleanups[i];
+    });
+    genericCleanup(this.tag);
+  }
+
+  run() {
+    this.sortInventoriesByCode();
+    this.addSortByCodeButton();
+  }
+
+
+
+  private sortInventoriesByCode() {
+    if (this.sortByCode) {
+      const inventories = Array.from(
+        document.querySelectorAll("div[class^='InventoryView__grid__'")
+      );
+
+      inventories.forEach(inventory => {
+        let sorted = Array.from(inventory.children).sort(function(
```

Comment:
i'd move the whole block into a dedicated function `sortIntentoryByCode(inventory)` and just call it here. That way the original function has a reduced complexity and we might even be able to add unit-tests for a single `sortIntentoryByCode(inventory)` call.